core/domain/user_domain.py

```diff
@@ -1481,3 +1481,89 @@ def to_dict(self) -> ExplorationUserDataDict:
             'most_recently_reached_checkpoint_state_name': (
                 self.most_recently_reached_checkpoint_state_name)
         }
+
+
+class LearnerGroupUserDict(TypedDict):
+    """Dictionary for LearnerGroupUser domain object."""
+
+    user_id: str
+    invited_to_learner_groups: List[str]
+    member_of_learner_groups: List[str]
+    progress_sharing_permissions: List[ProgressSharingPermissionsDict]
+
+
+class ProgressSharingPermissionsDict(TypedDict):
+    """Dictionary for Progress Sharing Permissions of learner groups."""
+
+    group_id: str
+    sharing_is_turned_on: bool
+
+
+class LearnerGroupUser:
+    """Domain object for learner group user."""
+
+    def __init__(
+        self,
+        user_id: str,
+        invited_to_learner_groups: List[str],
+        member_of_learner_groups: List[str],
+        progress_sharing_permissions: List[ProgressSharingPermissionsDict]
+    ) -> None:
+        """Constructs a LearnerGroupUser domain object.
+
+        Attributes:
+            user_id: str. The user id.
+            invited_to_learner_groups: list(str). List of learner group ids
+                that the user has been invited to.
+            member_of_learner_groups: list(str). List of learner group ids
+                that the user is a member of.
+            progress_sharing_permissions: list(ProgressSharingPermissionsDict).
```

Comment:
Shouldn't this be a domain object? We shouldn't be passing dicts around.

Reply:
Yeah, it should be. Updated this. Thanks!